src/tests/ftest/nvme/nvme_pool_extend.py

```diff
@@ -50,23 +49,20 @@ def run_nvme_pool_extend(self, num_pool, oclass=None):
             oclass (str) : object class (eg: RP_2G8,etc)
                            Defaults to None.
         """
-        pool = {}
+        self.pool = []
         total_servers = len(self.hostlist_servers) * 2
         self.log.info("Total Daos Servers (Initial): %d", total_servers)
         if oclass is None:
             oclass = self.ior_cmd.dfs_oclass.value
 
         for val in range(0, num_pool):
             # Create a pool
-            pool[val] = TestPool(self.context, dmg_command=self.dmg_command)
-            pool[val].get_params(self)
-            pool[val].create()
-            pool[val].set_property("reclaim", "disabled")
+            self.pool.append(self.get_pool())
+            self.pool[-1].set_property("reclaim", "disabled")
 
         # On each pool (max 3), extend the ranks
         # eg: ranks : 4,5 ; 6,7; 8,9.
         for val in range(0, num_pool):
-            self.pool = pool[val]
             test = self.ior_test_sequence[val]
             threads = []
             threads.append(threading.Thread(target=self.run_ior_thread,
```

Comment:
We need to pass in the kwarg `"pool": self.pool[val]` and OSAUtils.run_ior_thread() will need an optional `pool` argument which should default to self.pool if not set.